src/lib/y2packager/product.rb

```diff
@@ -0,0 +1,144 @@
+    # @return [String] Short name
+    attr_reader :short_name
+    # @return [String] Display name
+    attr_reader :display_name
+    # @return [String] Version
+    attr_reader :version
+    # @return [String] Architecture
+    attr_reader :arch
+    # @return [Symbol] Status
+    attr_reader :status
+    # @return [Symbol] Category
+    attr_reader :category
+    # @return [String] Vendor
+    attr_reader :vendor
+    # @return [Integer] Display order
+    attr_reader :order
+    # @return [String] package including installation.xml for install on top of lean os
+    attr_reader :installation_package
+
+    def self.available_base_products
+      Y2Packager::ProductReader.available_base_products
+    end
+
+    # Constructor
+    #
+    # @param name                 [String]  Name
+    # @param short_name           [String]  Short name
+    # @param display_name         [String]  Display name
+    # @param version              [String]  Version
+    # @param arch                 [String]  Architecture
+    # @param status               [Symbol]  Status (:selected, :removed, :installed, :available)
+    # @param category             [Symbol]  Category (:base, :addon)
+    # @param vendor               [String]  Vendor
+    # @param order                [Integer] Display order
+    # @param installation_package [String]  Installation package name
+    def initialize(name: nil, short_name: nil, display_name: nil, version: nil, arch: nil,
+      status: nil, category: nil, vendor: nil, order: nil, installation_package: nil)
+      @name = name
+      @short_name = short_name
+      @display_name = display_name
+      @version = version
+      @arch = arch.to_sym if arch
+      @status = status.to_sym if status
+      @category = category.to_sym if category
+      @vendor = vendor
+      @order = order
+      @installation_package = installation_package
+    end
+
+    # Compare two different products
+    #
+    # If arch, name, version and vendor match they are considered the
+    # same product.
+    #
+    # @return [Boolean] true if both products are the same; false otherwise
+    def ==(other)
+      result = arch == other.arch && name == other.name &&
+        version == other.version && vendor == other.vendor
+      log.info("Comparing products: '#{arch}' <=> '#{other.arch}', '#{name}' <=> '#{other.name}', "\
+        "'#{version}' <=> '#{other.version}', '#{vendor}' <=> '#{other.vendor}' => "\
+        "result: #{result}")
+      result
+    end
+
+    # is the product selected to install?
+    #
+    # @return [Boolean] true if it is selected
+    def selected?
+      Yast::Pkg.ResolvableProperties(name, :product, "").any? do |res|
+        res["status"] == :selected
+      end
+    end
+
+    # select the product to install
+    # @return [Boolean] true if the product has been sucessfully selected
+    def select
+      log.info "Selecting product #{name} to install"
+      Yast::Pkg.ResolvableInstall(name, :product, "")
```

Comment:
This is a bit tricky as the `Y2Packager::Product` object has also the `arch`, `version` and `vendor` attributes but this method ignores them.

So the question is if this API should be bound to the very specific libzypp product (as specified in the object attributes) or can be generic. In the original case we do not care about the version, we leave the responsibility on libzypp to select the highest version.

The API might be confusing, `Y2Packager::Product.new(name: "SLES", version: "15").select` might actually select a different version to install.

If this is OK (as it makes the implementation much easier) is should be at least documented in the method comment.